test/functional/discovered_hosts_controller_test.rb

```diff
@@ -84,27 +84,25 @@ def test_edit_form_build_mode_enabled
   end
 
   def test_edit_form_quick_submit
-    disable_taxonomies do
-      domain = FactoryBot.create(:domain)
-      hostgroup = FactoryBot.create(:hostgroup, :with_subnet, :with_environment, :with_rootpass, :with_os, :domain => domain)
-      new_ip = hostgroup.subnet.ipaddr
-      host = discover_host_from_facts(@facts.merge({
-                                                     'ipaddress' => new_ip,
-                                                     'ipaddress_eth0' => new_ip
-                                                   }))
-      get :edit, params: {
-        :id => host.id,
-        :quick_submit => true,
-        :host => {
-          :hostgroup_id => hostgroup.id
-        } }, session: set_session_user_default_manager
-
-      managed_host = Host.find(host.id)
-      assert managed_host.build
-      assert_redirected_to host_url(managed_host)
-      assert_equal hostgroup.id, managed_host.hostgroup_id
-      assert_match(/Successfully/, flash[:success])
-    end
+    domain = FactoryBot.create(:domain)
+    hostgroup = FactoryBot.create(:hostgroup, :with_subnet, :with_environment, :with_rootpass, :with_os, :domain => domain, organizations: domain.organizations, locations: domain.locations)
```

Comment:
I'm not sure setting org/loc here is needed as well, the factory default should be the same as the domain one.

Reply:
Yeah, earlier I had not set the loc/org here but still I was facing same failures. So I just gave this a try.